### PR TITLE
Documentation update with securitySchemes final agreement

### LINKED ISSUE #57: Align `securitySchemes` and `security` of CAMARA API specs with IdentityAndConsentManagement WG
This issue is intended to define a single common/standard OpenID Connect security scheme for CAMARA API specs aligned with the OIDC FE/BE flows defined in https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/user-consent.md#authorization-flows--grant-types as the access flows to CAMARA APIs.

e.g.
```
components:
  securitySchemes:
    openId:
      type: openIdConnect
      openIdConnectUrl: https://example.com/.well-known/openid-configuration
```

This ensures that the **CAMARA API specifications are consistent with each other and with the IdentityAndConsentManagement work**: 

- Any reference to other `securitySchemes`, such as client credentials, would generally be removed. Client credentials access (2-legged) must only be used _"when user resource is NOT involved & the client belongs to confidential category"_.
- Those APIs without defined scopes should define the corresponding scopes that protect the API operations, and they must be associated with the OpenID Connect security scheme by means of the `security` property.

**This has already been proposed in Commonalities in: https://github.com/camaraproject/Commonalities/issues/53 where the proposal is very detailed. So we can just refer this issue in IdentityAndConsentManagement to issue #53 in Commonalities. The same Commonalities issue is not only intended to unify the usage of `securitySchemes` and `security` fields, but also to define a common scope naming format for CAMARA API specifications.**

### PR DESCRIPTION
**Pull request by jpengar**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:

This PR is intended to include in the WG documentation the final agreement reached by the TSC and the WG participants on the `securitySchemes` and `security` of the CAMARA API specification, and the mandatory template for `info.description`, explaining to API customers why the API specification does not list specific grant types, and how to find out what authorization flows they can use.

#### Which issue(s) this PR fixes:

Fixes #57 

#### Changelog input

```
 CAMARA API Specification - Authorization and authentication common guidelines
```

#### Additional documentation 

N/A

### CODE REVIEW

**Review by diegogonmar — APPROVED** [review]
LGTM

File: documentation/CAMARA-API-access-and-user-consent.md [hunk 1]
```diff
@@ -354,3 +358,52 @@ If some usecase/s for an API point to off-net scenarios and where consumption an
...
+
+### Mandatory template for `info.description` in CAMARA API specs
+
+The documentation template below must be used as part of the API documentation in  `info.description` property in the CAMARA API specs:
+
+
+```
+### Authorization and authentication
+
+CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-API-access-and-user-consent.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md). Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
```
Comment (eric-murray) [inline documentation/CAMARA-API-access-and-user-consent.md:404]: Document "CAMARA-API-access-and-user-consent.md" says nothing about client credentials, and yet that remains a valid authorisation scheme for _some_ CAMARA API endpoints. When it was proposed that CAMARA adopt `type: openIdConnect` as the ONLY valid security scheme, this was justified on the basis that `.well-known/openid-configuration` allows `client_credentials` in the list of  `grant_types_supported`. And yet here we have a reference to a document that does not "outline" this as a possibility.

So for the proposed text to be applicable, "CAMARA-API-access-and-user-consent.md" must be updated to list client credentials as a valid authorisation scheme for some scenarios.

Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:404]: Agree. This is a fair point. We will commit an update to clarify that client credentials is a valid authorization scheme for some scenarios.

Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:411]: Done in commit https://github.com/camaraproject/IdentityAndConsentManagement/pull/93/commits/4f09ec6530261b425c560e337dd8d56dbfd15c0a

**Review by murthygorty — CHANGES_REQUESTED** [review]
In terms and concepts, can you consider changing "User" term to "Subscriber" please? If approved and changed, we will have 
1) End User -> human participant using the application (can be a parent or alternate delegate of the carrier-resource)
2) Subscriber -> subscriber of the telco operator, owner of the carrier-resource at hand. (eg., parent)

**Comment by shilpa-padgaonkar** [comment]
@jpengar:  Instead of replicating all info from AuthN-AuthZ doc again in this document,  may be it is easier to simply say that 

> CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-AuthN-AuthZ-Concept.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-AuthN-AuthZ-Concept.md). 

> 

All possible grant types are already documented in this doc.

**Comment by jpengar** [comment]
>In terms and concepts, can you consider changing "User" term to "Subscriber" please? If approved and changed, we will have
>
>End User -> human participant using the application (can be a parent or alternate delegate of the carrier-resource)
>Subscriber -> subscriber of the telco operator, owner of the carrier-resource at hand. (eg., parent)

@murthygorty This specific topic is not related to the original issue #57 or the content included in this PR. Would you kindly open a new issue to discuss this specific topic? thank you so much in advance.
I always encourage WG participants to keep the original scope of the discussion to avoid creating unnecessary dependencies, to keep the discussion focused on the topic, and to prevent the discussion from becoming deadlocked.

**Comment by jpengar** [comment]
>@jpengar: Instead of replicating all info from AuthN-AuthZ doc again in this document, may be it is easier to simply say that
>
>CAMARA guidelines defines a set of authorization flows which can grant API clients access to the API functionality, as outlined in the document [CAMARA-AuthN-AuthZ-Concept.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-AuthN-AuthZ-Concept.md).
>
>All possible grant types are already documented in this doc.

@shilpa-padgaonkar This document actually mentions additional grant types/flows to those considered and defined as a valid option to access CAMARA APIs: Auth Code Flow, CIBA, and Client Credentials.

**Comment by shilpa-padgaonkar** [comment]
@jpengar : That is indeed correct. My main concern is that we will soon get requests to cover more options here if not explicitly specified for eg. there might be a request to add details about PKCE or other possible extensions.

Another option would be that we keep the link as is, in your PR, but mention that details about possible extensions etc. can be found in the AuthN-AuthZ doc.

Also, for recommended flows under oauth2, just like client credentials we should also specify auth code flow.

**Comment by jpengar** [comment]
@shilpa-padgaonkar Could you use the github.com change suggestion feature to add your suggestion to the current status of the PR? That way we avoid a trail and error process in case I don't fully understand your point. :S

File: documentation/CAMARA-API-access-and-user-consent.md [hunk 2]
```diff
@@ -18,6 +18,11 @@ This document defines guidelines for telco operator exposure platforms to manage
     - [Authorization flows / grant types](#authorization-flows--grant-types)
       - [Authorization code flow (Frontend flow)](#authorization-code-flow-frontend-flow)
```
Comment (shilpa-padgaonkar) [inline documentation/CAMARA-API-access-and-user-consent.md:19]:
```suggestion
Detailed information on the below specified grant types (for eg. extensions to some of the grant types, links to original RFC documents etc.) is available in [CAMARA-AuthN-AuthZ-Concept.md](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-AuthN-AuthZ-Concept.md)
      - [Authorization code flow (Frontend flow)](#authorization-code-flow-frontend-flow)
```

Comment (jpengar) [reply documentation/CAMARA-API-access-and-user-consent.md:19]: Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/93/commits/d20fe33f6f8c02db75f6475acb18cd65aae8c986

**Comment by jpengar** [comment]
@shilpa-padgaonkar @rartych @bigludo7 @eric-murray As agreed in the 20/12 WG meeting call, now that the PR has already addressed your change suggestions, could you do a final review to merge the PR and close the issue?

**Review by shilpa-padgaonkar — APPROVED** [review]
/LGTM

**Review by hdamker — APPROVED** [review]

**Review by rartych — APPROVED** [review]
LGTM

**Review by AxelNennker — APPROVED** [review]
LGTM

**Comment by murthygorty** [comment]
@jpengar 
about [this](>https://github.com/camaraproject/IdentityAndConsentManagement/pull/93#issuecomment-1864123852), you are 100% right, PR comments should only be about the issue at hand. Thanks, will follow #98

**Pull request merged**